test/NormalizingStorageAdapter.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace test\eLife\Journal;
+
+use Csa\Bundle\GuzzleBundle\Cache\StorageAdapterInterface;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Uri;
+use GuzzleHttp\Psr7\UriNormalizer;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Throwable;
+use function GuzzleHttp\json_decode;
+use function GuzzleHttp\json_encode;
+
+final class NormalizingStorageAdapter implements StorageAdapterInterface
+{
+    const URI_FLAGS = UriNormalizer::PRESERVING_NORMALIZATIONS | UriNormalizer::SORT_QUERY_PARAMETERS;
+
+    private $storageAdapter;
+
+    public function __construct(StorageAdapterInterface $storageAdapter)
+    {
+        $this->storageAdapter = $storageAdapter;
+    }
+
+    public function fetch(RequestInterface $request)
+    {
+        return $this->storageAdapter->fetch($this->normalize($request));
+    }
+
+    public function save(RequestInterface $request, ResponseInterface $response)
+    {
+        $this->storageAdapter->save($this->normalize($request), $response);
+    }
+
+    private function normalize(RequestInterface $request) : RequestInterface
+    {
+        $headers = array_change_key_case($request->getHeaders());
```

Comment:
Need to finish off https://github.com/csarrazi/CsaGuzzleBundle/pull/202 as it's an upstream bug.